### PR TITLE
Adding kubeconfig as pre requisite in BUILD.md

### LINKED ISSUE #1543: [Docs] Specify K8s cluster requirement as a prerequisite for running tests
`make test` requires Kubernetes cluster config in the environment. Add this as a requirement in `BUILD.md`.

### PR DESCRIPTION
**Pull request by Sagar2366**
## Change Overview

Adding kubeconfig as pre requisite in BUILD.md

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #1543 

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ The [Makefile](Makefile) provides a set of targets to help simplify the build
 tasks. To ensure cross-platform consistency, many of these targets use Docker
 to spawn build containers based on the `ghcr.io/kanisterio/build` public image.
 
+By default, helm looks for a file named config in the $HOME/.kube directory. You can specify other kubeconfig files by setting the KUBECONFIG environment variable or by setting the --kubeconfig flag.
+Make sure the Kuberntes cluster config is present in $HOME/.kube directory for next comamnds to successfully execute.
+
```
Comment (ihcsim) [inline BUILD.md:37]: Please wrap each line to 80 characters, per the rest of the documentation.

> By default, helm looks for a file

`make test` isn't using Helm. I think all we need to say here is that for `make test` to succeed, a valid kubeconfig file must be found at `$HOME/.kube/config`. See the Docker command that runs `make test` [here](https://github.com/kanisterio/kanister/blob/fa04d77eb6f5c92521d1413ddded385168f39f42/Makefile#L219).

File: helm/kanister-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -15,6 +16,7 @@ spec:
 {{ include "kanister-operator.helmLabels" . | indent 8}}
     spec:
       serviceAccountName: {{ template "kanister-operator.serviceAccountName" . }}
+      automountServiceAccountToken: false
```
Comment (ihcsim) [inline helm/kanister-operator/templates/deployment.yaml:19]: Not needed.

File: helm/fdb-operator/templates/manager/statefulset.yaml [hunk 3]
```diff
@@ -24,6 +24,7 @@ spec:
           secret:
             secretName: {{ .Values.secret.name }}
       serviceAccountName: default
+      automountServiceAccountToken: false
```
Comment (ihcsim) [inline helm/fdb-operator/templates/manager/statefulset.yaml:27]: Not needed.

File: helm/fdb-operator/templates/manager/statefulset.yaml [hunk 4]
```diff
@@ -63,4 +63,4 @@ spec:
         - containerPort: 9876
           name: webhook-server
           protocol: TCP
-      terminationGracePeriodSeconds: 10
\ No newline at end of file
+      terminationGracePeriodSeconds: 10
```
Comment (ihcsim) [inline helm/fdb-operator/templates/manager/statefulset.yaml:66]: Not needed.

Comment (pavannd1) [reply helm/fdb-operator/templates/manager/statefulset.yaml:66]: I think it is needed but not related to this PR

File: helm/kanister-operator/templates/deployment.yaml [hunk 5]
```diff
@@ -2,6 +2,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ template "kanister-operator.fullname" . }}
+  namespace: {{ template "kanister-operator.namespace" . }}
```
Comment (pavannd1) [inline helm/kanister-operator/templates/deployment.yaml:5]: Same as above, this is an unrelated change - does not belong in this PR

File: BUILD.md [hunk 6]
```diff
@@ -32,6 +32,9 @@ The [Makefile](Makefile) provides a set of targets to help simplify the build
 tasks. To ensure cross-platform consistency, many of these targets use Docker
 to spawn build containers based on the `ghcr.io/kanisterio/build` public image.
 
+For `make test` to succeed, a valid kubeconfig file must be found at 
+`$HOME/.kube/config`. See the Docker command that runs make test [here](https://github.com/kanisterio/kanister/blob/fa04d77eb6f5c92521d1413ddded385168f39f42/Makefile#L219).
```
Comment (pavannd1) [inline BUILD.md:36]:
```suggestion
For `make test` to succeed, a valid `kubeconfig` file must be found at 
`$HOME/.kube/config`. See the Docker command that runs `make test` [here](https://github.com/kanisterio/kanister/blob/fa04d77eb6f5c92521d1413ddded385168f39f42/Makefile#L219).
```

**Comment by Sagar2366** [comment]
@ihcsim @pavannd1 opened separate PR - https://github.com/kanisterio/kanister/pull/1663.
Can we close this one ?
